ra_aid/agent_utils.py

```
@@ -528,44 +711,68 @@ def run_agent_with_retry(agent, prompt: str, config: dict) -> Optional[str]:
+                    InternalServerError,
+                    APITimeoutError,
+                    RateLimitError,
+                    APIError,
+                    ValueError,
+                ) as e:
+                    if isinstance(e, ValueError):
+                        error_str = str(e).lower()
+                        if "code" not in error_str or "429" not in error_str:
+                            raise  # Re-raise ValueError if it's not a Lambda 429
                     if attempt == max_retries - 1:
                         logger.error("Max retries reached, failing: %s", str(e))
```

Ruff (0.8.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)